sway/config/seat.c

```diff
@@ -0,0 +1,132 @@
+#define _XOPEN_SOURCE 700
+#include <stdlib.h>
+#include <string.h>
+#include "sway/config.h"
+#include "log.h"
+
+struct seat_config *new_seat_config(const char* name) {
+	struct seat_config *seat = calloc(1, sizeof(struct seat_config));
+	if (!seat) {
+		sway_log(L_DEBUG, "Unable to allocate seat config");
+		return NULL;
+	}
+
+	sway_log(L_DEBUG, "new_seat_config(%s)", name);
+	seat->name = strdup(name);
+	if (!sway_assert(seat->name, "could not allocate name for seat")) {
+		return NULL;
+	}
+
+	seat->fallback = -1;
+	seat->attachments = create_list();
+	if (!sway_assert(seat->attachments,
+				"could not allocate seat attachments list")) {
```

Comment:
should free seat->name and seat here (free_seat_config isn't safe to use here because it doesn't check seat->attachments before using it)